node/pkg/txverifier/evm.go

```diff
@@ -181,12 +181,101 @@ func (tv *TransferVerifier[ethClient, Connector]) pruneCache() {
 	}
 
 	if numPrunedEvals > 0 {
-		tv.logger.Debug("pruned cached receipt evaluations",
+		tv.logger.Info("pruned receipt evaluations cache",
 			zap.Int("numPruned", numPrunedEvals),
 			zap.Int("cacheSize", len(tv.evaluations)),
 		)
 	}
-	// TODO: The other caches should be pruned here too.
+
+	// For the other caches, just prune them if they're too big.
+	if len(tv.chainIdCache) >= CacheMaxSize {
+		var (
+			i = 0
+			// Slice to store keys to delete in order to avoid deleting keys from maps that are being
+			// iterated over.
+			keysToDelete = make([]string, CacheDeleteCount)
```

Comment:
Instead of having the delete count being a fixed number, should the count be `max(CacheDeleteCount, len(tv.chainIdCache) - CacheMaxSize)`? Or simply the latter, where you trim to `CacheMaxSize`? Otherwise this could theoretically grow indefinitely?

Reply:
My idea was basically like "if this gets to 100 or above, just remove 10 entries". (arbitrary numbers)
How would this grow indefinitely?

Reply:
This only fires if there are over `CacheMaxSize` (100) entries in the cache though, e.g. see L213. And when it does, it only removes 10 entries. I'm not sure how the scenario you're imaging could happen.